src/saml2/saml.py

```diff
@@ -191,6 +192,11 @@ def _wrong_type_value(xsd, value):
                 'to_type': _str,
                 'to_text': _str,
             },
+            'date': {
+                'type': datetime.date,
+                'to_type': lambda x: datetime.datetime.strptime(x, '%Y-%m-%d').date(),
```

Comment:
This is setting a specific time format, whereas `xsd:date` allows for `[-]CCYY-MM-DD[Z\|(+\|-)hh:mm]`. I'll probably do #518 first, then point this to the new datetime parser (`saml2.datetime.parse`) that takes that into account as well as timezones.

Reply:
I agree with you. That code was useful just for date string parsing, a better generalization on top of #518 would be the best solution for taking also datetime as timestamps, with timezone support.

Reply:
Date format do not need timezone or minutes and seconds. Bytheway if you want to parse date and datetime together with a single function... Why not.